chartfx-dataset/src/test/java/de/gsi/dataset/spi/DoubleErrorDataSetTests.java

```diff
@@ -0,0 +1,222 @@
+package de.gsi.dataset.spi;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import static de.gsi.dataset.DataSet.DIM_X;
+import static de.gsi.dataset.DataSet.DIM_Y;
+
+import java.util.Arrays;
+
+import org.junit.jupiter.api.Test;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Checks for DoubleErrorDataSet interfaces and constructors.
+ * 
+ * @author rstein
+ */
+public class DoubleErrorDataSetTests extends EditableDataSetTests {
+    private static final Logger LOGGER = LoggerFactory.getLogger(DoubleErrorDataSetTests.class);
+    protected static final double[][] testCoordinate = { { 1.0, 2.0, 3.0 }, { 2.0, 4.0, 6.0 } };
+    protected static final double[] testEYZERO = { 0.0, 0.0, 0.0 };
+    protected static final double[] testEYN = { 0.2, 0.3, 0.4 };
+    protected static final double[] testEYP = { 0.1, 0.2, 0.3 };
+    private static final int n = testCoordinate[0].length;
+
+    @Test
+    public void defaultTests() {
+        // re-check EditableDataSet interface consistency
+        EditableDataSetTests.checkEditableDataSetInterface(new DoubleErrorDataSet("test"));
+
+        DoubleErrorDataSet firstDataSet = new DoubleErrorDataSet("test");
+        checkAddPoints(firstDataSet, 0); // w/o errors
+
+        final DoubleErrorDataSet secondDataSetA = new DoubleErrorDataSet("test", testCoordinate[0], testCoordinate[1],
+                testEYZERO, testEYZERO, n, true);
+        assertEquals(firstDataSet, secondDataSetA, "DoubleErrorDataSet(via arrays, deep copy) constructor");
+
+        final DoubleErrorDataSet secondDataSetB = new DoubleErrorDataSet("test", Arrays.copyOf(testCoordinate[0], n),
+                Arrays.copyOf(testCoordinate[1], n), Arrays.copyOf(testEYZERO, n), Arrays.copyOf(testEYZERO, n), n,
+                false);
+        assertEquals(firstDataSet, secondDataSetB, "DoubleErrorDataSet(via arrays, no deep copy) constructor");
+
+        checkAddPoints(firstDataSet, 1); // with errors but w/o label and style
+
+        checkAddPoints(firstDataSet, 2); // with errors and label but w/o style
+
+        checkAddPoints(firstDataSet, 3); // with errors and empty label but w/o style
+
+        checkAddPoints(firstDataSet, 4); // with errors (array version)
+
+        checkAddPoints(firstDataSet, 5); // with errors (array version)
+
+        final DoubleErrorDataSet thirdDataSet = new DoubleErrorDataSet(firstDataSet);
+        assertEquals(firstDataSet, thirdDataSet, "DoubleErrorDataSet(DataSet2D) constructor");
+
+        assertNotEquals(0, firstDataSet.getDataCount(), "pre-check clear method");
+        firstDataSet.clearData();
+        assertEquals(0, firstDataSet.getDataCount(), "check clear method");
+    }
+
+    public static void checkAddPoints(final DoubleErrorDataSet dataSet, final int testCase) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [The method 'checkAddPoints(DoubleErrorDataSet, int)' has an NPath complexity of 260](https://app.codacy.com/manual/GSI/chart-fx/pullRequest?prid=4728036)